catalyst_voices/packages/catalyst_voices_shared/lib/src/common/build_environment.dart

```diff
@@ -0,0 +1,23 @@
+enum BuildEnvironment {
+  dev,
+  qa,
+  preprod,
+  prod;
+
+  /// Converts a `String` value to a corresponding `BuildEnvironment`
+  /// enum value.
+  static String fromEnvironment(String value) {
```

Comment:
The docs for this method say it will return enum but here it's a string.

What do you think about adding a field to an enum called tag and iterating over all values?
Because this runs on strings the compiler will not enforce switch exhaustiveness. In my opinion following pattern, not only here but for all enums which are mappable to a string, works better:

```dart
enum BuildEnvironment {
dev(tag: 'dev');

final String tag;
const BuildEnvironment({required this.tag});

static BuildEnvironment fromEnvironment(String string) {
  for (final value in values) {
    if (value.tag == string) return value;
  }
  
  // nothing found, return default:
  return dev;
}
}
```

Reply:
I updated the docs. I'd suggest keeping values pre-defined and not dynamic atm.